apis/v1alpha1/shared_types.go

```diff
@@ -194,6 +194,37 @@ type RouteGatewayStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// GatewayStatusReference identifies a Gateway in a specified namespace. This
+// reference also includes a controller name to simplify cleaning up status
+// entries.
+type GatewayStatusReference struct {
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Namespace is the namespace of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Namespace string `json:"namespace"`
+
+	// Controller is a domain/path string that indicates the controller
+	// implementing the Gateway. This corresponds with the controller field on
+	// GatewayClass.
+	//
+	// Example: "acme.io/gateway-controller".
+	//
+	// The format of this field is DOMAIN "/" PATH, where DOMAIN and PATH are
+	// valid Kubernetes names
+	// (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).
+	//
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
```

Comment:
Should we make this required?
Conformant implementations will set this field and clean up after themselves. But then this API allows for a mistake whereby an implementation might not set the controller field and then the user has a sub-optimal experience.

Reply:
I want to make this required in v1alpha2, but I don't think we can require it in v0.3.0 or else it would be a breaking change. I'll make a tracking issue for this if/when this PR gets in.

Reply:
Ah, I missed that. Makes sense.